### PR TITLE
mat64: simplify offset calculation

### PR DESCRIPTION
**Pull request by kortschak**
This comes from discussion in the go issue raised by the offset code.

@btracey Please take a look.

### CODE REVIEW

**Comment by btracey** [comment]
What's the issue number again? 

(Can't take a good look until later tonight)

**Comment by kortschak** [comment]
12445

File: mat64/offset.go [hunk 1]
```diff
@@ -10,17 +10,11 @@ import "unsafe"
...
+	if &a[0] == &b[0] {
 		return 0
 	}
-	if a0 < b0 {
-		return int((b0 - a0) / unsafe.Sizeof(float64(0)))
-	}
-	return -int((a0 - b0) / unsafe.Sizeof(float64(0)))
+	// This expression must be atomic with respect to GC moves.
+	// At this stage this is true, because the GC does not
+	// move.
```
Comment (btracey) [inline mat64/offset.go]: Maybe add a "See golang/go issue 12445"?

**Comment by btracey** [comment]
LGTM

**Pull request merged**